fineract-loan/src/main/java/org/apache/fineract/portfolio/interestpauses/service/InterestPauseReadPlatformServiceImpl.java

```diff
@@ -0,0 +1,156 @@
+import java.util.List;
+import java.util.Locale;
+import lombok.RequiredArgsConstructor;
+import org.apache.fineract.infrastructure.core.domain.ExternalId;
+import org.apache.fineract.infrastructure.core.service.DateUtils;
+import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
+import org.apache.fineract.portfolio.interestpauses.data.InterestPauseData;
+import org.apache.fineract.portfolio.loanaccount.data.LoanTermVariationsData;
+import org.apache.fineract.portfolio.loanaccount.domain.Loan;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanRepository;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanTermVariationType;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanTermVariations;
+import org.apache.fineract.portfolio.loanaccount.rescheduleloan.domain.LoanTermVariationsRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class InterestPauseReadPlatformServiceImpl implements InterestPauseReadPlatformService {
+
+    private final PlatformSecurityContext context;
+    private final LoanTermVariationsRepository loanTermVariationsRepository;
+    private final LoanRepository loanRepository;
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<InterestPauseData> retrieveInterestPauses(Long loanId) {
+        this.context.authenticatedUser();
+
+        List<LoanTermVariationsData> variations = this.loanTermVariationsRepository.findLoanTermVariationsByLoanIdAndTermType(loanId,
+                LoanTermVariationType.INTEREST_PAUSE.getValue());
+
+        return mapToInterestPauseData(variations);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<InterestPauseData> retrieveInterestPausesByExternalId(String loanExternalId) {
+        this.context.authenticatedUser();
+
+        List<LoanTermVariationsData> variations = this.loanTermVariationsRepository.findLoanTermVariationsByExternalLoanIdAndTermType(
+                new ExternalId(loanExternalId), LoanTermVariationType.INTEREST_PAUSE.getValue());
+
+        return mapToInterestPauseData(variations);
+    }
+
+    @Override
+    @Transactional
+    public Long createInterestPause(ExternalId loanExternalId, String startDate, String endDate, String dateFormat, String locale) {
+        this.context.authenticatedUser();
+
+        Long loanId = loanRepository.findIdByExternalId(loanExternalId);
+        if (loanId == null) {
+            throw new IllegalArgumentException("Loan not found with External ID: " + loanExternalId);
+        }
+
+        Loan loan = loanRepository.findById(loanId).orElseThrow(() -> new IllegalArgumentException("Loan not found with ID: " + loanId));
+
+        validateInterestPauseDates(loan, startDate, endDate, dateFormat, locale);
+
+        LocalDate startLocalDate = parseAndValidateDate(startDate, dateFormat, locale);
```

Comment:
you can parse from command the LocalDate or by using the DateUtils.parseLocalDate()`

Reply:
Fixed